chitchat/src/failure_detector.rs

```diff
@@ -55,23 +41,30 @@ impl FailureDetector {
                     self.config.max_interval,
                     self.config.initial_interval,
                 )
-            });
+            })
+            .report_heartbeat();
     }
 
     /// Marks the node as dead or alive based on the current phi value.
     pub fn update_node_liveness(&mut self, chitchat_id: &ChitchatId) {
-        if let Some(phi) = self.phi(chitchat_id) {
-            debug!(node_id=%chitchat_id.node_id, phi=phi, "updating node liveness");
-            if phi > self.config.phi_threshold {
-                self.live_nodes.remove(chitchat_id);
+        let phi_opt = self.phi(chitchat_id);
+        let is_alive = self
+            .phi(chitchat_id)
+            .map(|phi| phi <= self.config.phi_threshold)
+            .unwrap_or(false);
+        debug!(node_id=%chitchat_id.node_id, phi=?phi_opt, is_alive=is_alive, "computing node liveness");
+        if is_alive {
+            self.live_nodes.insert(chitchat_id.clone());
+            self.dead_nodes.remove(chitchat_id);
+        } else {
+            self.live_nodes.remove(chitchat_id);
+            if !self.dead_nodes.contains_key(&chitchat_id) {
```

Comment:
this was a bug.
When a node was identified twice as dead, we would update its time of death.